randomwallpaper@iflow.space/wallpaperController.js

```diff
@@ -193,6 +193,9 @@ var WallpaperController = class {
 		 https://bitbucket.org/LukasKnuth/backslide/src/7e36a49fc5e1439fa9ed21e39b09b61eca8df41a/backslide@codeisland.org/settings.js?at=master
 		 */
 		if (settings.is_writable("picture-uri")) {
+			// Set a picture options:
+			let stretching = this._settings.get('enable-image-stretching', 'boolean');
+			settings.set_string("picture-options", stretching ? 'spanned' : 'zoom');
```

Comment:
```suggestion
			if (stretching) {
				settings.set_string('picture-options', 'spanned');
			}
```

This would override the current setting that a user has for the desktop every time the image is modified.
I'd suggest only adjusting the setting when stretching is enabled. Otherwise, leave it as it is.

Reply:
E.g. a user might want to use something different than 'zoom' or 'spanned' (there are more options)

Reply:
Do you think it worth adding options list like in schema?
https://gitlab.gnome.org/GNOME/gsettings-desktop-schemas/-/blob/master/schemas/org.gnome.desktop.screensaver.gschema.xml.in#L88-95

Reply:
All of these options are dynamically supported by the experimental feature that I mentioned b1ca7086b414cfddf6e31e057dc825d3b83d2aa7.  

Again this only supports GS 40, and if you are interested, you could take a look at it and backport it.
For your current PR, I do not see it as necessary.

Reply:
The only thing I would ensure is that a manual setting is not changed by the extension if image stretching is disabled.